src/core/Handler.cpp

```diff
@@ -310,14 +309,12 @@ void CServerHandler::onRequest(const Pistache::Http::Request& req, Pistache::Htt
         }
 
         // attempt to handle mime
-        magic_t magic = magic_open(MAGIC_MIME_TYPE);
-        if (magic && magic_load(magic, nullptr) == 0) {
+        if (magic_t magic = magic_open(MAGIC_MIME_TYPE); magic_load(magic, nullptr) == 0) {
             const char* m        = magic_file(magic, PATH_ABSOLUTE.c_str());
             auto        mimeType = Pistache::Http::Mime::MediaType::fromString(m ? std::string(m) : std::string("application/octet-stream"));
             response.headers().add<Pistache::Http::Header::ContentType>(mimeType);
         }
-        if (magic)
-            magic_close(magic);
+        if(magic) magic_close(magic);
```

Comment:
this will not build? magic is a local within the if above. Just make the magic before the if, then if load succeeds do the stuff, then close if the ptr is not null